File: 전예진/Greedy/prob6.py
def solution(routes):
    answer = 0
    
    # 차량의 진출 지점을 기준으로 정렬
    routes.sort(key=lambda x: x[1])
    
    # 카메라 위치 초기화
    camera = -30001
    
    for route in routes:
        if camera < route[0]:
            # 카메라가 현재 차량의 진입 지점보다 이전에 있으면
            # 현재 차량을 만나지 않았으므로 카메라 설치
            camera = route[1]
            answer += 1
    
    return answer

File: 주다애/Greedy/prob6.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        List<Car> list = new ArrayList<>();
        for(int[] route : routes) {
            list.add(new Car(route[0], route[1]));
        }
      // end시점으로 오름차순 정렬하는 것이 포인트++
        Collections.sort(list);
        int end = list.get(0).getEnd();
        
        for(int i = 1; i < list.size(); i++) {
            if(list.get(i).getStart() <= end) continue;
            end = list.get(i).getEnd();
            answer++;
        }
        
        return answer+1;
    }
}

class Car implements Comparable<Car> {
    int start;
    int end;
    
    Car(int start, int end) {
        this.start = start;
        this.end = end;
    }
    
    public int getStart() {
        return this.start;
    }
    
    public int getEnd() {
        return this.end;
    }
    
    public int compareTo(Car c) {
        if(this.end != c.end) {
            return this.end - c.end;
        }
        return this.start - c.start;
    }
}
